config/src/lua.rs

```diff
@@ -71,19 +71,36 @@ pub fn make_lua_context(config_file: &Path) -> anyhow::Result<Lua> {
                 }
             }
         }
+        let config_file_str = config_file
+            .to_str()
+            .ok_or_else(|| anyhow!("config file path is not UTF-8"))?;
 
-        wezterm_mod.set(
-            "config_file",
-            config_file
-                .to_str()
-                .ok_or_else(|| anyhow!("config file path is not UTF-8"))?,
-        )?;
+        lua.set_named_registry_value("wezterm-watch-paths", config_file_str)?;
+        wezterm_mod.set("config_file", config_file_str)?;
         wezterm_mod.set(
             "config_dir",
             config_dir
                 .to_str()
                 .ok_or_else(|| anyhow!("config dir path is not UTF-8"))?,
         )?;
+        wezterm_mod.set(
+            "add_to_config_reload_watch_list",
+            lua.create_function(|lua, args: Variadic<String>| {
+                let watch_path: String = lua.named_registry_value("wezterm-watch-paths")?;
```

Comment:
Rather than splitting/joining strings, let's just make this an array:

```suggestion
                let mut watch_paths: Vec<String> = lua.named_registry_value("wezterm-watch-paths")?;
```

Reply:
does it automatically add dividers or what?